### PR TITLE
Added stork volume snapshot test triggers

### PR DESCRIPTION
**Pull request by gejain**
Added stork volume snapshot test triggers

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PTX-24966

**Special notes for your reviewer**:
Refer trigger run result - https://aetos.pwx.purestorage.com/resultSet/testSetID/705387/testCaseID/4563771

### CODE REVIEW

File: tests/testTriggers.go [hunk 1]
```diff
@@ -10283,6 +10287,132 @@ func TriggerAsyncDRMigrationSchedule(contexts *[]*scheduler.Context, recordChan
...
+		if err != nil {
+			UpdateOutcome(event, fmt.Errorf("Failed to get pvc list for namespace"))
+			return
+		}
+		err = asyncdr.ValidateSnapshotScheduleCount(pvcNames, ns, schdPol.Name, snapshotType, snapInterval, retain)
+		if err != nil {
+			UpdateOutcome(event, fmt.Errorf("Failed to validate snapshot schedule count"))
+			return
+		}
+	}
```
Comment (kphalgun-px) [inline tests/testTriggers.go:10350]: It would be helpful while debugging if we print params in the log messages.

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/testTriggers.go (hunk 1 above)
Comment (gejain) [reply tests/testTriggers.go]: Added log for PVC and NS. Thanks!

File: pkg/asyncdr/asyncdr.go [hunk 2]
```diff
@@ -281,6 +282,117 @@ func CreateMigrationSchedule(
...
+					PersistentVolumeClaimName: pvcName},
+			},
+			SchedulePolicyName: policyName,
+		},
+	}
+	snapSchedule, err := storkops.Instance().CreateSnapshotSchedule(snapSched)
+	return snapSchedule, err
+}
+
+func CreateSchedulePolicyWithRetain(policyName string, interval int, retain storkapi.Retain) (pol *storkapi.SchedulePolicy, err error) {
```
Comment (Rohit-PX) [inline pkg/asyncdr/asyncdr.go:312]: TODO later: Just have a  generic function called `CreateSchedulePolicy` and based on if `retain != nil` , use it to create SchedulePolicy with or without `retain`.
In my opinion we don't need a function just to create a snapshot will retain policy.

File: tests/testTriggers.go [hunk 3]
```diff
@@ -10283,6 +10287,134 @@ func TriggerAsyncDRMigrationSchedule(contexts *[]*scheduler.Context, recordChan
...
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	var (
+		taskNamePrefix     = "snaptest"
```
Comment (Rohit-PX) [inline tests/testTriggers.go:10375]:
```suggestion
		taskNamePrefix     = "stork-snaptest"
```

**Review by Rohit-PX — APPROVED** [review]
Except these two comments, looks good to me.

File: pkg/asyncdr/asyncdr.go (hunk 2 above)
Comment (gejain) [reply pkg/asyncdr/asyncdr.go:312]: Yes, as there are lot of tests using it, I will refactor this later.

File: tests/testTriggers.go (hunk 3 above)
Comment (gejain) [reply tests/testTriggers.go]: Done

**Pull request merged**